pkg/scale/decode_test.go

```diff
@@ -535,3 +538,85 @@ func Test_decodeState_decodeUint(t *testing.T) {
 		})
 	}
 }
+
+type myStruct struct {
```

Comment:
Since one of the use cases is decoding into a struct with an interface as a field type, perhaps could be nice to have as a test case